### PR TITLE
Authenticate user api

### PR DESCRIPTION
**Pull request by rebeccasedgwick**
Adds User authentication: json web tokens used to check calls to api.

Create & register (signup) routes to api remain available without tokens.

### CODE REVIEW

**Comment by rebeccasedgwick** [comment]
Updated: changed routes for clarity
`auth/login` to `/login`
`/signup` to `register`

**Review by EsamAl-Dabagh — APPROVED** [review]

File: app/controllers/login_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class LoginController < ApplicationController
+  skip_before_action :authorize_request, only: :authenticate
+
+  def authenticate
+    auth_token = AuthenticateUser.new(auth_params[:email], auth_params[:password]).call
+    json_response(auth_token: auth_token)
+  end
+
```
Comment (fredericolawson) [inline app/controllers/login_controller.rb:8]: @rebeccasedgwick @leekarho just a q - how come this is not a RESTful action?

File: config/routes.rb [hunk 2]
```diff
@@ -6,4 +6,7 @@
   resources :mentees
   resources :mentorships
 
+  post 'login', to: 'login#authenticate'
+  post 'register', to: 'users#create'
+
```
Comment (fredericolawson) [inline config/routes.rb:11]: @rebeccasedgwick @leekarho we don't think we need `post 'register'` - we already have a route to the `users#create` action

File: app/controllers/login_controller.rb (hunk 1 above)
Comment (rebeccasedgwick) [reply app/controllers/login_controller.rb:8]: For readability for humans -- have reverted it now to User#create.

**Review by fredericolawson — APPROVED** [review]

**Review by fredericolawson — APPROVED** [review]

**Pull request merged**